### PR TITLE
Upgrade softflowd used by crawler to extended version and use Netflow V10

### PR DESCRIPTION
**Pull request by stefanberger**
Upgrade the softflowd used by the crawler's fprobe plugin to a newer and extended version
and use Netflow V10 (IPFIX) instead of the previous Netflow V5. The benefit of IPFIX and the extended softflowd is that it now can detect which side initiated a TCP connection (not UDP or ICMP, though). We can use this information on the backend side to annotate initiators of a connection.

Signed-off-by: Stefan Berger <stefanb@linux.vnet.ibm.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -5,10 +5,15 @@ WORKDIR /crawler
...
+COPY \
+  dependencies/python-socket-datacollector_0.1.1-1_all.deb \
+  dependencies/softflowd_0.9.9901-1_amd64.deb \
+  /tmp/
 RUN dpkg -i /tmp/python-socket-datacollector_*_all.deb && \
-    apt-get update && \
-    DEBIAN_FRONTEND=noninteractive apt-get -y install softflowd
+    apt-get -y update && \
+    apt-get -y install libpcap0.8 && \
+    dpkg -i /tmp/softflowd_0.9.*amd64.deb && \
```
Comment (sahilsuneja1) [inline Dockerfile]: why not apt install any more?

File: crawler/plugins/systems/fprobe_container_crawler.py [hunk 2]
```diff
@@ -145,6 +145,8 @@ def start_fprobe(self, ifname, user, bindaddr, port, **kwargs):
                   '-n', '%s:%d' % (bindaddr, port)]
         if len(fprobe_bpf.strip()):
             params.insert(1, fprobe_bpf)
+        if netflow_version == 10:
+            params.insert(1, '-b')
```
Comment (sahilsuneja1) [inline crawler/plugins/systems/fprobe_container_crawler.py]: is this a standard flag? or works with the specific softflowd version part of this PR?

File: Dockerfile [hunk 3]
```diff
@@ -5,10 +5,15 @@ WORKDIR /crawler
 COPY requirements.txt /crawler/requirements.txt
 RUN pip install -r requirements.txt
 
-COPY dependencies/python-socket-datacollector_0.1.1-1_all.deb /tmp
+COPY \
+  dependencies/python-socket-datacollector_0.1.1-1_all.deb \
+  dependencies/softflowd_0.9.9901-1_amd64.deb \
```
Comment (sahilsuneja1) [inline Dockerfile]: the file submitted with PR is a *.tar.xz

File: Dockerfile (hunk 1 above)
Comment (stefanberger) [inline Dockerfile]: Switching to an extended version that's not available via Ubuntu.

File: crawler/plugins/systems/fprobe_container_crawler.py (hunk 2 above)
Comment (stefanberger) [inline crawler/plugins/systems/fprobe_container_crawler.py]: This starts working with the extended version of softflowd.

Comment (stefanberger) [inline crawler/plugins/systems/fprobe_container_crawler.py]: I now removed the source code file from the dependencies directory.

**Comment by sahilsuneja1** [comment]
codecov seems unhappy at test coverage of these changes. This blocking behaviour seems new. Do you wanna tackle this given the new flow probes you are currently working on?

**Comment by stefanberger** [comment]
I modified the test case a bit. Maybe we get some of the lost coverage back.

**Comment by stefanberger** [comment]
Codecov is good now. There was one build that failed for non-obivous reasons.

**Pull request merged**